cst438_registration/src/test/java/com/cst438/JunitTestStudent.java

```diff
@@ -0,0 +1,204 @@
+                found = true;
+            }
+        }
+        assertEquals(true, found, "Added student not in updated student database.");
+
+        // Check case where same email is entered
+        response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .post("/student/add")
+                                .content(asJsonString(studentTest))
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(406, response.getStatus());
+
+    }
+
+    @Test
+    @DirtiesContext
+    public void deleteStudent() throws Exception {
+        MockHttpServletResponse response;
+
+        // Test deleting student with existing enrollment
+        response = mvc.perform(
+                MockMvcRequestBuilders
+                        .delete("/student/delete?id=1")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(400, response.getStatus());
+
+        // Test force delete on student with enrollment
+        response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .delete("/student/delete?id=1&FORCE=true")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(200, response.getStatus());
+
+        // Verify student no longer exists
+        response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .get("/student/list")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        Student[] students = fromJsonString(response.getContentAsString(), Student[].class);
+        boolean found = false;
+        for( Student s : students) {
+            if(s.getStudent_id() == 1) {
+                found = true;
+            }
+        }
+        assertFalse(found);
+
+        // Test delete on student with no enrollment
+        response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .delete("/student/delete?id=4")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(200, response.getStatus());
+
+        for( Student s : students) {
+            if(s.getStudent_id() == 1) {
+                found = true;
+            }
+        }
+        assertFalse(found);
+    }
+
+    @Test
```

Comment:
The unit tests for each CRUD method are all very comprehensive and easy to read as well. The spacing, coding conventions, and consistency all help in making it readable. The comments are very efficient as well, they are not overly complicated, but not too vague as well. Overall I approve of this request.